core/rawdb/ancient_utils.go

```diff
@@ -50,36 +50,58 @@ func (info *freezerInfo) size() common.StorageSize {
 	return total
 }
 
+func inspect(name string, order map[string]bool, reader ethdb.AncientReader) (freezerInfo, error) {
+	info := freezerInfo{name: name}
+	for t := range order {
+		size, err := reader.AncientSize(t)
+		if err != nil {
+			return freezerInfo{}, err
+		}
+		info.sizes = append(info.sizes, tableSize{name: t, size: common.StorageSize(size)})
+	}
+	// Retrieve the number of last stored item
+	ancients, err := reader.Ancients()
+	if err != nil {
+		return freezerInfo{}, err
+	}
+	info.head = ancients - 1
+
+	// Retrieve the number of first stored item
+	tail, err := reader.Tail()
+	if err != nil {
+		return freezerInfo{}, err
+	}
+	info.tail = tail
+	return info, nil
+}
+
 // inspectFreezers inspects all freezers registered in the system.
 func inspectFreezers(db ethdb.Database) ([]freezerInfo, error) {
 	var infos []freezerInfo
 	for _, freezer := range freezers {
 		switch freezer {
 		case chainFreezerName:
-			// Chain ancient store is a bit special. It's always opened along
-			// with the key-value store, inspect the chain store directly.
-			info := freezerInfo{name: freezer}
-			// Retrieve storage size of every contained table.
-			for table := range chainFreezerNoSnappy {
-				size, err := db.AncientSize(table)
-				if err != nil {
-					return nil, err
-				}
-				info.sizes = append(info.sizes, tableSize{name: table, size: common.StorageSize(size)})
+			info, err := inspect(chainFreezerName, chainFreezerNoSnappy, db)
+			if err != nil {
+				return nil, err
+			}
+			infos = append(infos, info)
+
+		case stateFreezerName:
+			datadir, err := db.AncientDatadir()
```

Comment:
Are you sure this is correct? This seems to be using the default location, which may or may not be the same as teh one actually used. IMO it might make sense to support having the state history freezer at a custom path since I could want to put ancients on HDD but state history maybe on SSD if I run an archive node?

Reply:
Make sense. 

But it will introduce a lot complexity. Basically we need to duplicate all the flags for state history (e.g. `datadir.ancient-state`?)

Now I just use `datadir.ancient` as a root directory and all freezers(not only chain and state, I guess we will have trace freezer or something else in the future) are a subfolder inside of it.

Do you think it makes sense that we just keep the state history inside of the ancient root directory as now, and provide CLI flags only when we ship the archive node?